backend/src/environment.ts

```diff
@@ -63,13 +67,48 @@ export class Environment {
     this.isElevated = await isElevated()
   }
 
-  get manufactureId() {
+  get manufactureId(): number {
     return this.isHeadless ? MANUFACTURE_ID_HEADLESS : MANUFACTURE_ID_STANDARD
   }
 
   get frontend() {
+    let osVersion = os.release()
+    if (this.isMac) {
+      try {
+        let versionInfo: any = plist.parse(fs.readFileSync('/System/Library/CoreServices/SystemVersion.plist', 'utf8'))
+        console.log(JSON.stringify(versionInfo.ProductVersion))
+        osVersion = versionInfo.ProductVersion
+      } catch {
+        console.log('No System Version File: /System/Library/CoreServices/SystemVersion.plist')
+      }
+    }
+
+    if (this.manufacturerDetails === undefined) {
+      try {
+        let manufactererFile = JSON.parse(fs.readFileSync(path.join(this.adminPath, 'manufacturer.json'), 'utf8'))
```

Comment:
What happens if file doesn't exist? What happens if content isn't in the format we expect?

Reply:
then the catch occurs and we load defaults